docs/content/latest/develop/ecosystem-integrations/apache-spark/java.md

```diff
@@ -37,105 +36,130 @@ isTocNested: true
 
 </ul>
 
-## Before you begin
-
-### Maven
+## Setting Up the Project with Maven
 
-To build your Java application using the YugabyteDB Spark Connector for YCQL, add the following snippet to your `pom.xml` for Scala 2.11:
+To build a Java application using the YugabyteDB Spark Connector for YCQL, add the following snippet to your `pom.xml` file for Scala 2.12:
 
 ```xml
 <dependency>
   <groupId>com.yugabyte.spark</groupId>
-  <artifactId>spark-cassandra-connector_2.11</artifactId>
-  <version>2.4-yb-3</version>
+  <artifactId>spark-cassandra-connector_2.12</artifactId>
+  <version>3.0-yb-1</version>
 </dependency>
 ```
 
-## Run a sample application
+For more information, see [Maven artifact](https://search.maven.org/artifact/com.yugabyte.spark/spark-cassandra-connector_2.12).
 
-### Run the Spark word-count sample application
+## Using Apache Spark with YCQL
 
-You can run our Spark-based sample application with:
+Suppose you work with a YCQL table created as follows: 
 
-```sh
-$ java -jar yb-sample-apps.jar --workload CassandraSparkWordCount --nodes 127.0.0.1:9042
+```sql
+CREATE TABLE test.person (
+  id int PRIMARY KEY,
+  name text,
+  address text,
+  phone jsonb
+);
 ```
 
-It reads data from a table with sentences — by default, it generates an input table `ybdemo_keyspace.lines`, computes the frequencies of the words, and writes the result to the output table `ybdemo_keyspace.wordcounts`.
-
-### Examine the source code
-
-To look at the source code, you can check:
+This table is populated with the following rows:
 
-- the source file in our GitHub source repo [here](https://github.com/yugabyte/yugabyte-db/blob/master/java/yb-loadtester/src/main/java/com/yugabyte/sample/apps/CassandraSparkWordCount.java)
-- untar the jar `java/yb-sample-apps-sources.jar` in the download bundle
+```
+id  | name  | address                | phone  
+----+-------+------------------------+-----------------------------
+1   | John  | Hammersmith London, UK | {"code":"+44","phone":1000}
+2   | Nick  | Acton London, UK       | {"code":"+43","phone":1200}
+3   | Smith | 11 Acton London, UK    | {"code":"+44","phone":1400}
+```
 
-Most of the logic is in the `run()` method of the `CassandraSparkWordCount` class (in the file `src/main/java/com/yugabyte/sample/apps/CassandraSparkWordCount.java`). Some of the key portions of the sample program are explained in the sections below.
+The following Java code shows how to initialize a Spark session:
 
-## Main sections of an Apache Spark program on Yugabyte
+```java
+// Setup the local spark master
+SparkConf conf = new SparkConf().setAppName("yb.spark-jsonb")
+  .setMaster("local[1]")
+  .set("spark.cassandra.connection.localDC", "datacenter1")
+  .set("spark.cassandra.connection.host", addresses.**get**(0).getHostName())
```

Comment:
Remove the `**`, looks like a copy paste issue maybe:
```
set("spark.cassandra.connection.host", addresses.get(0).getHostName())
```

Reply:
Fixed